documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,182 @@
+
+### Revocation Bitmap Service
+
+In order to allow verfiers to check the status of a verifiable credential, the DID document of the issuer may contain a service that MUST have the following properties:
+
+- **id**: a URL identifying the revocation bitmap;
+- **type**: the string `"RevocationBitmap2022"`;
+- **serviceEndpoint**: a `data` URL [[RFC2397](https://datatracker.ietf.org/doc/html/rfc2397)] embedding a revocation bitmap generated according to the [bitmap generation algorithm](#Bitmap-Generation-Algorithm). The `<mediatype>` MUST be `application/octet-stream` and the `base64` attribute MUST be set.
+
+## Algorithms
+
+The following algorithms describe how to generate and validate revocation bitmaps.
+
+### Bitmap Generation Algorithm
+
+The following process MUST be followed when producing a `RevocationBitmap2022`:
+
+1. Let **bitmap** be a [_roaring bitmap_](https://roaringbitmap.org/) where each bit is initialized to 0 (zero).
+2. For each revoked credential with an **identifier** not exceeding an unsigned, 32-bit integer, set the bit in **bitmap** at index **identifier** to 1.
+3. Generate the **bitmap serialization** according to the [roaring bitmap serialization format](https://github.com/RoaringBitmap/RoaringFormatSpec/) using the **bitmap** as input.
+4. Generate a **compressed bitmap** by using the ZLIB compression algorithm [[RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)] on the bitmap serialization and base64-encoding [[RFC4648](https://datatracker.ietf.org/doc/html/rfc4648)] the result.
+5. Return the **compressed bitmap**.
+
+### Bitmap Expansion Algorithm
+
+The following process MUST be followed when expanding a compressed revocation list bitmap.
+
+1. Let **compressed bitmap** be a compressed revocation bitmap generated using the above algorithm.
+2. Generate an **uncompressed bitmap** by base64-decoding [[RFC4648](https://datatracker.ietf.org/doc/html/rfc4648)] the **compressed bitmap** and further decompressing using ZLIB [[RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)].
+3. Generate the **bitmap** by deserializing the **uncompressed bitmap** according to the [roaring bitmap serialization format](https://github.com/RoaringBitmap/RoaringFormatSpec/).
+4. Return the **bitmap**.
+
+### Validation Algorithm
+
+The following steps MUST be followed when checking whether a verifiable credential is revoked:
+
+1. Let **credential** be a verifiable credential containing a `credentialStatus` whose `type` is `RevocationBitmap2022`.
+2. Let **revocation bitmap uri** be the `id` field of the **credential**'s `credentialStatus`.
+3. Verify all proofs associated with the **credential**. If a proof fails, return a validation error.
+4. Resolve the **revocation bitmap uri** to a **revocation bitmap service**, and verify that its type is `RevocationBitmap2022`. Return an error otherwise.
```

Comment:
```suggestion
4. Resolve the **revocation bitmap URI** to a **revocation bitmap service** in the issuer's DID Document, and verify that the service `type` is `RevocationBitmap2022`. Return an error otherwise.
```